### PR TITLE
libtiff: add libtiff/4.4.0

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libtiff/4.4.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libtiff/all/conanfile.py [hunk 1]
```diff
@@ -185,24 +183,25 @@ def package(self):
...
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
         self.cpp_info.set_property("cmake_find_mode", "both")
         self.cpp_info.set_property("cmake_file_name", "TIFF")
         self.cpp_info.set_property("cmake_target_name", "TIFF::TIFF")
-        self.cpp_info.set_property("pkg_config_name", "libtiff-{}".format(tools.Version(self.version).major))
+        self.cpp_info.set_property("pkg_config_name", "libtiff-{}".format(Version(self.version).major))
```
Comment (SpaceIm) [inline recipes/libtiff/all/conanfile.py:193]:
```suggestion
        self.cpp_info.set_property("pkg_config_name", "libtiff-{}".format(Version(self.version).major))
```

There is no `major` accessor in `conan.tools.scm.Version`, it returns `None`.

File: recipes/libtiff/all/conanfile.py [hunk 2]
```diff
@@ -185,24 +183,25 @@ def package(self):
...
+        if self.options.shared and self.settings.os == "Windows" and not is_msvc(self):
             self.cpp_info.libs = [lib + ".dll" for lib in self.cpp_info.libs]
         if self.settings.os in ["Linux", "Android", "FreeBSD", "SunOS", "AIX"]:
             self.cpp_info.system_libs.append("m")
 
         # TODO: to remove in conan v2 once cmake_find_package* & pkg_config generators removed
         self.cpp_info.names["cmake_find_package"] = "TIFF"
         self.cpp_info.names["cmake_find_package_multi"] = "TIFF"
-        self.cpp_info.names["pkg_config"] = "libtiff-{}".format(tools.Version(self.version).major)
+        self.cpp_info.names["pkg_config"] = "libtiff-{}".format(Version(self.version).major)
```
Comment (SpaceIm) [inline recipes/libtiff/all/conanfile.py:207]: same comment here, major returns None

File: recipes/libtiff/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/libtiff/all/conanfile.py]: `major` is a method of `Version`, so it would print e.g. `<bound method Version.major of '3.4'>`

Comment (SpaceIm) [reply recipes/libtiff/all/conanfile.py]: It is not, conans.tools.Version != conan.tools.scm.Version ;)

Comment (madebr) [reply recipes/libtiff/all/conanfile.py]: I am saying `major` is also available on `conan.tools.scm.Version`, not as a property, but as a method.

Run the following in your terminal:
```sh
python -c "import conan.tools.scm; v = conan.tools.scm.Version('5.4.3.2'); print(f'{v.major=}')"
```

Comment (SpaceIm) [reply recipes/libtiff/all/conanfile.py]: Indeed

Comment (SpaceIm) [reply recipes/libtiff/all/conanfile.py]: so something like this would work: `Version(self.version).major(fill=False)`

Comment (madebr) [reply recipes/libtiff/all/conanfile.py]: I changed it with `version[0]`, that looked simpler to me.

**Review by uilianries — DISMISSED** [review]

File: recipes/libtiff/all/conanfile.py [hunk 3]
```diff
@@ -1,9 +1,11 @@
-from conan.tools.files import rename
+from conan.tools.files import get, rename, rmdir
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
 from conans import ConanFile, CMake, tools
 from conans.errors import ConanInvalidConfiguration
 import os
 
-required_conan_version = ">=1.43.0"
+required_conan_version = ">=1.46.0"
```
Comment (SpaceIm) [inline recipes/libtiff/all/conanfile.py:8]:
```suggestion
required_conan_version = ">=1.47.0"
```

Due to `rmdir`

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/libtiff/all/conanfile.py [hunk 4]
```diff
@@ -185,24 +183,25 @@ def package(self):
...
+        if self.options.shared and self.settings.os == "Windows" and not is_msvc(self):
             self.cpp_info.libs = [lib + ".dll" for lib in self.cpp_info.libs]
         if self.settings.os in ["Linux", "Android", "FreeBSD", "SunOS", "AIX"]:
             self.cpp_info.system_libs.append("m")
 
         # TODO: to remove in conan v2 once cmake_find_package* & pkg_config generators removed
         self.cpp_info.names["cmake_find_package"] = "TIFF"
         self.cpp_info.names["cmake_find_package_multi"] = "TIFF"
-        self.cpp_info.names["pkg_config"] = "libtiff-{}".format(tools.Version(self.version).major)
+        self.cpp_info.names["pkg_config"] = "libtiff-{}".format(Version(self.version)[0])
```
Comment (SpaceIm) [inline recipes/libtiff/all/conanfile.py:207]: To fix once https://github.com/conan-io/conan/pull/11823 released in a version we can use in cci?

Comment (uilianries) [reply recipes/libtiff/all/conanfile.py:207]: As soon as we have 1.52.0 or 1.51.2 running on CCI.

**Pull request merged**